src/pybind/mgr/orchestrator/module.py

```diff
@@ -1188,6 +1188,46 @@ def _apply_rgw(self,
                 out = to_format(data, format, many=True, cls=None)
         return HandleCommandResult(stdout=out)
 
+    @_cli_write_command(
+        'orch apply haproxy',
+        'name=placement,type=CephString,req=false '
+        'name=dry_run,type=CephBool,req=false '
+        'name=format,type=CephChoices,strings=plain|json|json-pretty|yaml,req=false '
+        'name=unmanaged,type=CephBool,req=false',
+        'Update the number of HAProxy instances')
+    def _apply_haproxy(self,
+                   placement: Optional[str] = None,
+                   dry_run: bool = False,
+                   format: str = 'plain',
+                   unmanaged: bool = False,
+                   inbuf: Optional[str] = None) -> HandleCommandResult:
+
+        usage = """Usage:
+  ceph orch apply haproxy -i <yaml spec> [--dry-run]
```

Comment:
As you has defined the command the usage will be: (cahnaged name and removed things not needed)
ceph orch apply ha_rgw -i <yaml spec> 
or 
ceph orch apply ha_rgw --placement [host1, ... hostn]

Reply:
i think there are 2 options here. just using `ceph orch apply ha_rgw --placement [host1, ... hostn]` would not be sufficient we need all the other info too so i can add all these arguments 
![image](https://user-images.githubusercontent.com/22037319/97341720-d1730a00-185b-11eb-89a5-da84dc41bce5.png)
 or i can remove the placement argument and we only accept a spec file.

Reply:
for now i removed the placement argument and we only accept a spec file but let me know what you think?

Reply:
Is ok. Maybe we can use the placement and use default values chosen by us for the rest of options, but maybe it is not worth. So it is ok !